src/aklite_client_ext.cc

```diff
@@ -0,0 +1,146 @@
+#include "aktualizr-lite/aklite_client_ext.h"
+
+#include <sys/file.h>
+#include <unistd.h>
+#include <boost/format.hpp>
+#include <boost/process.hpp>
+#include <boost/property_tree/ini_parser.hpp>
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include "http/httpclient.h"
+#include "libaktualizr/config.h"
+#include "liteclient.h"
+#include "primary/reportqueue.h"
+
+#include "aktualizr-lite/tuf/tuf.h"
+#include "composeapp/appengine.h"
+#include "composeappmanager.h"
+#include "docker/restorableappengine.h"
+#include "ostree/repo.h"
+#include "tuf/akhttpsreposource.h"
+#include "tuf/akrepo.h"
+#include "tuf/localreposource.h"
+#include "uptane/exceptions.h"
+
+// TODO: Reuse this
+static Uptane::Target toUptaneTarget(const TufTarget& t) {
+  Json::Value target_json;
+  target_json["hashes"]["sha256"] = t.Sha256Hash();
+  target_json["custom"]["targetFormat"] = "OSTREE";
+  target_json["length"] = 0;
+  Uptane::Target target(t.Name(), target_json);
+  return target;
+}
+
+// Returns the target that should be installed, if any.
+// It might be an updated version, a rollback target, or even the currently installed target, in case we need to sync
+// apps
+CheckInResult AkliteClientExt::GetTargetToInstall(const LocalUpdateSource* local_update_source, int version,
+                                                  const std::string& target_name, const TufTarget& current,
+                                                  bool allow_bad_target, bool force_apps_sync) {
+  // Tell CheckIn and CheckInLocal methods to not call notifyTufUpdateFinished in case of success
+  usingUpdateClientApi = true;
+  client_->setAppsNotChecked();
+  bool rollback_operation = false;
+
+  CheckInResult res = local_update_source == nullptr ? CheckIn() : CheckInLocal(local_update_source);
+
+  if (res.status != CheckInResult::Status::Ok && res.status != CheckInResult::Status::OkCached) {
+    LOG_WARNING << "Unable to update latest metadata";
+    return res;
+  }
+
+  auto candidate_target = res.SelectTarget(version, target_name);
```

Comment:
Just an idea. Maybe instead of `SelectTarget(version, target_name)` we can have two methods `cr.GetTarget(int version)` and `cr.GetTarget(string name)`. Passing these two params looks a bit awkward. And we have these two params in  `GetTargetToInstall` too...

Maybe we can encapsulate these two params in something like `struct TargetId { int Version; string Name}`?